diff -c -r -N binutils-2.27/bfd/config.bfd binutils-2.27_new/bfd/config.bfd
*** binutils-2.27/bfd/config.bfd	2016-08-03 09:36:50.000000000 +0200
--- binutils-2.27_new/bfd/config.bfd	2018-05-05 14:52:36.754643707 +0200
***************
*** 671,676 ****
--- 671,679 ----
      targ_selvecs="iamcu_elf32_vec i386_aout_linux_vec i386_pei_vec"
      targ64_selvecs="x86_64_elf64_vec x86_64_elf32_vec x86_64_pei_vec l1om_elf64_vec k1om_elf64_vec"
      ;;
+   i[3-7]86-*-ctOS)
+     targ_defvec=i386_elf32_vec
+     ;;
    i[3-7]86-*-nacl*)
      targ_defvec=i386_elf32_nacl_vec
      targ_selvecs="arm_elf32_nacl_be_vec arm_elf32_nacl_le_vec"
diff -c -r -N binutils-2.27/config.sub binutils-2.27_new/config.sub
*** binutils-2.27/config.sub	2016-08-03 09:36:51.000000000 +0200
--- binutils-2.27_new/config.sub	2018-05-05 14:30:00.532965555 +0200
***************
*** 1533,1538 ****
--- 1533,1541 ----
  		;;
  	-ios)
  		;;
+ 	-ctOS*)
+         	os=-ctOS
+         	;;
  	-none)
  		;;
  	*)
diff -c -r -N binutils-2.27/gas/configure.tgt binutils-2.27_new/gas/configure.tgt
*** binutils-2.27/gas/configure.tgt	2016-08-03 09:36:51.000000000 +0200
--- binutils-2.27_new/gas/configure.tgt	2018-05-05 14:56:51.858016086 +0200
***************
*** 217,222 ****
--- 217,223 ----
    i386-*-beospe*)			fmt=coff em=pe ;;
    i386-*-beos*)				fmt=elf ;;
    i386-*-coff)				fmt=coff ;;
+   i386-*-ctOS)                		fmt=elf ;;
    i386-*-elfiamcu)			fmt=elf arch=iamcu ;;
    i386-*-elf*)				fmt=elf ;;
    i386-*-kaos*)				fmt=elf ;;
diff -c -r -N binutils-2.27/ld/configure.tgt binutils-2.27_new/ld/configure.tgt
*** binutils-2.27/ld/configure.tgt	2016-08-03 09:36:54.000000000 +0200
--- binutils-2.27_new/ld/configure.tgt	2018-05-05 15:28:37.784956482 +0200
***************
*** 316,321 ****
--- 316,322 ----
  			targ_extra_emuls=elf_i386 ;;
  i[3-7]86-*-elf*)	targ_emul=elf_i386
  			targ_extra_emuls=elf_iamcu ;;
+ i[3-7]86-*-ctOS)    	targ_emul=elf_ctOS ;;
  x86_64-*-elf* | x86_64-*-rtems*)
  			targ_emul=elf_x86_64
  			targ_extra_emuls="elf_i386 elf_iamcu elf32_x86_64 elf_l1om elf_k1om"
diff -c -r -N binutils-2.27/ld/emulparams/elf_ctOS.sh binutils-2.27_new/ld/emulparams/elf_ctOS.sh
*** binutils-2.27/ld/emulparams/elf_ctOS.sh	1970-01-01 01:00:00.000000000 +0100
--- binutils-2.27_new/ld/emulparams/elf_ctOS.sh	2018-05-05 15:32:29.036307540 +0200
***************
*** 0 ****
--- 1,16 ----
+ . ${srcdir}/emulparams/plt_unwind.sh
+ SCRIPT_NAME=elf
+ OUTPUT_FORMAT="elf32-i386"
+ NO_RELA_RELOCS=yes
+ TEXT_START_ADDR=0x40000000
+ MAXPAGESIZE="CONSTANT (MAXPAGESIZE)"
+ COMMONPAGESIZE="CONSTANT (COMMONPAGESIZE)"
+ ARCH=i386
+ MACHINE=
+ NOP=0x90909090
+ TEMPLATE_NAME=elf32
+ GENERATE_SHLIB_SCRIPT=no
+ GENERATE_PIE_SCRIPT=no
+ NO_SMALL_DATA=yes
+ SEPARATE_GOTPLT=12
+ IREL_IN_PLT=
diff -c -r -N binutils-2.27/ld/Makefile.in binutils-2.27_new/ld/Makefile.in
*** binutils-2.27/ld/Makefile.in	2016-08-03 09:36:54.000000000 +0200
--- binutils-2.27_new/ld/Makefile.in	2018-05-05 15:50:03.557523182 +0200
***************
*** 655,660 ****
--- 655,661 ----
  	eelf32xstormy16.c \
  	eelf32xtensa.c \
  	eelf_i386.c \
+ 	eelf_ctOS.c \
  	eelf_i386_be.c \
  	eelf_i386_chaos.c \
  	eelf_i386_fbsd.c \
***************
*** 2900,2905 ****
--- 2901,2909 ----
  eelf_i386.c: $(srcdir)/emulparams/elf_i386.sh \
    $(ELF_X86_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
  
+ eelf_ctOS.c: $(srcdir)/emulparams/elf_ctOS.sh \
+   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+ 
  eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
    $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
  
